Signal/src/textsecure/Messages/TSMessagesManager.m

```diff
@@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
 }
 
 - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
+    if(content.group!= nil )  {
+        __block BOOL ignoreMessage = NO;
+        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
+            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
+            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
+            if(gThread==nil) {
+                ignoreMessage = YES;
+            }
+        }];
+        if(ignoreMessage) {
+            DDLogDebug(@"recevied message from group that I left, ignoring");
```

Comment:
so open for discussion here, but we should have consistent behavior with
android. according to the understanding (Moxie wasn't there)

a) that is exactly what it should be doing in the case of the group create
message not yet being received while group messages are already being
received (wait for the create and let the server handle resending the
messages buffered while I waiting)
b) in the case of me receiving messages in a group I have left I agree this
should maybe be changed. however the server should do the right thing and
if the client sending me the messages in the group I have left hasn't acked
my group I have left message the server will continue to send "Christine
left the group" until they do, not requiring me to send another "I left the
group" message myself. which I could also do, but I think that would result
in more traffic not less.


On Fri, Jan 2, 2015 at 8:54 PM, Frederic Jacobs <notifications@github.com>
wrote:

> In Signal/src/textsecure/Messages/TSMessagesManager.m
> <https://github.com/WhisperSystems/Signal-iOS/pull/248#discussion-diff-22422473>
> :
>
> > @@ -179,6 +179,21 @@ - (void)handlePreKeyBundle:(IncomingPushMessageSignal*)preKeyMessage{
> >  }
> >
> >  - (void)handleIncomingMessage:(IncomingPushMessageSignal*)incomingMessage withPushContent:(PushMessageContent*)content{
> > +    if(content.group!= nil )  {
> > +        __block BOOL ignoreMessage = NO;
> > +        [self.dbConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
> > +            GroupModel *emptyModelToFillOutId = [[GroupModel alloc] initWithTitle:nil memberIds:nil image:nil groupId:content.group.id]; // TODO refactor the TSGroupThread to just take in an ID (as it is all that it uses). Should not take in more than it uses
> > +            TSGroupThread *gThread = [TSGroupThread threadWithGroupModel:emptyModelToFillOutId transaction:transaction];
> > +            if(gThread==nil) {
> > +                ignoreMessage = YES;
> > +            }
> > +        }];
> > +        if(ignoreMessage) {
> > +            DDLogDebug(@"recevied message from group that I left, ignoring");
>
> What? What do you mean by not ack? Not acking on a network level? That
> sounds like a pretty bad idea since the server will just buffer those ones
> up.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WhisperSystems/Signal-iOS/pull/248/files#r22422473>.
>



-- 
Dr. Corbett Moran
Rockets @ http://www.spacex.com
Physics @ http://www.ICS.uzh.ch  <http://www.ICS.uzh.ch>
Dev @ http://circleof6app.com
Dev @ https://whispersystems.org
www.christinecorbettmoran.com